### PR TITLE
bitlbee service: Add option to load libpurple plugins into bitlbee

### PR DESCRIPTION
**Pull request by Nadrieril**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/networking/bitlbee.nix [hunk 1]
```diff
@@ -25,6 +29,13 @@ let
     ${cfg.extraDefaults}
     '';
 
+  purple_plugin_path =
+    builtins.concatStringsSep ":" (
```
Comment (joachifm) [inline nixos/modules/services/networking/bitlbee.nix]: Consider using `concatMapStringsSep`

Comment (Nadrieril) [inline nixos/modules/services/networking/bitlbee.nix]: Done, thx for the suggestion

**Comment by Mic92** [comment]
@makefu @4z3 are you interested in testing this?

**Comment by makefu** [comment]
@Mic92 will do! I will test telegram-purple and see how far i can get

**Comment by Nadrieril** [comment]
BTW I've been using it with purple-hangouts for a few months and it works very well.

**Comment by makefu** [comment]
@Mic92 @Nadrieril i've tested the module (via cherry-picking on my private nixpkgs unstable) with telegram-purple and everything works as expected. Thanks a lot, i was waiting for this feature for a long time.

**Comment by Nadrieril** [comment]
All the thanks go to @regnat, he was the one who wrote the code. I'm just trying to merge it since he is not interested anymore.

**Pull request merged**